src/app/archive/archive_lib/processor.ml

```diff
@@ -93,152 +108,191 @@ module Snarked_ledger_hash = struct
+            ; amount=
+                Signed_command.amount t
+                |> Core.Option.map ~f:Currency.Amount.to_int
+            ; fee= Signed_command.fee t |> Currency.Fee.to_int
+            ; memo= Signed_command.memo t |> Signed_command_memo.to_string
+            ; hash= transaction_hash |> Transaction_hash.to_base58_check
+            ; status= None
+            ; failure_reason= None }
+
+    let add_with_status (module Conn : CONNECTION) (t : Signed_command.t)
+        (status : User_command_status.t) =
+      let open Deferred.Result.Let_syntax in
+      let%bind user_command_id = add_if_doesn't_exist (module Conn) t in
+      let ( status_str
+          , failure_reason
+          , fee_payer_account_creation_fee_paid
           , receiver_account_creation_fee_paid
-          , created_token )
-        , user_command_id )
-    in
-    user_command_id
+          , created_token ) =
+        match status with
+        | Applied
+            { fee_payer_account_creation_fee_paid
+            ; receiver_account_creation_fee_paid
+            ; created_token } ->
+            let amount_to_int64 x =
+              Unsigned.UInt64.to_int64 (Currency.Amount.to_uint64 x)
+            in
+            ( "applied"
+            , None
+            , Option.map ~f:amount_to_int64 fee_payer_account_creation_fee_paid
+            , Option.map ~f:amount_to_int64 receiver_account_creation_fee_paid
+            , Option.map created_token ~f:(fun tid ->
+                  Unsigned.UInt64.to_int64 (Token_id.to_uint64 tid) ) )
+        | Failed failure ->
+            ( "failed"
+            , Some (User_command_status.Failure.to_string failure)
+            , None
+            , None
+            , None )
+      in
+      let%map () =
+        Conn.exec
+          (Caqti_request.exec
+             Caqti_type.(
+               tup3
+                 (tup2 (option string) (option string))
+                 (tup3 (option int64) (option int64) (option int64))
+                 int)
+             "UPDATE user_commands \n\
+              SET status = ?, \n\
+             \    failure_reason = ?, \n\
+             \    fee_payer_account_creation_fee_paid = ?, \n\
+             \    receiver_account_creation_fee_paid = ?, \n\
+             \    created_token = ? \n\
+              WHERE id = ?")
+          ( (Some status_str, failure_reason)
+          , ( fee_payer_account_creation_fee_paid
+            , receiver_account_creation_fee_paid
+            , created_token )
+          , user_command_id )
+      in
+      user_command_id
+  end
+
+  let as_user_command (t : User_command.t) : Coda_base.Signed_command.t =
+    match t with
+    | Signed_command c ->
+        c
+    | Snapp_command c ->
+        let module S = Coda_base.Snapp_command in
+        let ({source; receiver; amount} : S.transfer) = S.as_transfer c in
+        let fee_payer = S.fee_payer c in
+        { signature= Signature.dummy
```

Comment:
Can we still disambiguate between normal signed_command payments and snap commands coerced into payments via the `typ` column in the db? Could you put some metadata here in order to recover that?